### PR TITLE
PWN-5587 - NameService. Feature toggles and into app screen integration

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5587

## Description of Work

- feature toggles
- some stuff is duck taped (new key in shared prefs for example)
- fixed onboarding restore/create with network_config
- !!! BEFORE MERGING CHECK THAT DEVNET IS NOT DEFAULT !!!

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/appsfly/AppsFlyerService.kt [hunk 1]
```diff
@@ -19,7 +18,7 @@ class AppsFlyerService(private val context: Context) {
     fun install(application: Application, devKey: String) {
         val appsFlyer = AppsFlyerLib.getInstance()
         kotlin.runCatching {
-            appsFlyer.setDebugLog(BuildConfig.DEBUG)
+            appsFlyer.setDebugLog(false) // turn to true if needed
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/appsfly/AppsFlyerService.kt:21]: feature flag?)😉

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/appsfly/AppsFlyerService.kt (hunk 1 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/appsfly/AppsFlyerService.kt:21]: nah, it's so verbose, all the logcat in this stuff

File: app/src/main/java/org/p2p/wallet/auth/ui/reserveusername/OnboardingReserveUsernameFragment.kt [hunk 2]
```diff
@@ -31,24 +36,82 @@ class OnboardingReserveUsernameFragment :
...
+    }
+
+    private fun initQaSkipInstrument() {
+        binding.imageViewBanner.setOnClickListener {
+            clicksBeforeDebugSkip--
+            if (clicksBeforeDebugSkip == 0) {
                 close()
-                return@setOnMenuItemClickListener true
+            } else {
+                showUiKitSnackBar("Are you clicking me to skip the username..?! I dare you to click again! 🤬")
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/auth/ui/reserveusername/OnboardingReserveUsernameFragment.kt:96]: Looks like you forgot smth

File: app/src/main/java/org/p2p/wallet/infrastructure/network/environment/NetworkEnvironmentManager.kt [hunk 3]
```diff
@@ -57,11 +59,12 @@ class NetworkEnvironmentManager(
...
-            NetworkEnvironment.RPC_POOL
-        } else {
-            availableNetworks.first()
-        }
+        return NetworkEnvironment.DEVNET
+//        return if (NetworkEnvironment.RPC_POOL in availableNetworks) {
+//            NetworkEnvironment.RPC_POOL
+//        } else {
+//            availableNetworks.first()
+//        }
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/infrastructure/network/environment/NetworkEnvironmentManager.kt:67]: Forgot to revert

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/auth/ui/reserveusername/OnboardingReserveUsernameFragment.kt (hunk 2 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/auth/ui/reserveusername/OnboardingReserveUsernameFragment.kt:96]: nope, it's for the QA to skip this step

File: app/src/main/java/org/p2p/wallet/infrastructure/network/environment/NetworkEnvironmentManager.kt (hunk 3 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/infrastructure/network/environment/NetworkEnvironmentManager.kt:67]: yessss, thanks

**Pull request merged**